dissect/target/plugins/apps/webserver/nginx.py

```diff
@@ -1,97 +1,354 @@
+            for log_name in self.ERROR_LOG_NAMES:
+                self.error_paths.update(log_dir.glob(f"{log_name}*"))
+
+        seen = set()
 
         # Check for custom paths in NGINX install config
-        if (config_file := self.target.fs.path("/etc/nginx/nginx.conf")).exists():
-            for line in config_file.open("rt"):
-                line = line.strip()
-                if not line or "access_log " not in line:
-                    continue
+        for config_file in self.DEFAULT_CONFIG_PATHS:
+            if "*" in config_file:
+                base, _, glob = config_file.partition("*")
+                for f in self.target.fs.path(base).rglob(f"*{glob}"):
+                    if f not in seen:
+                        self.parse_config(f)
+                        seen.add(f)
 
-                try:
-                    line = line.split("access_log")[1].strip()
-                    log_path = self.target.fs.path(line.split()[0])
-                    log_paths.extend(
-                        path for path in log_path.parent.glob(f"{log_path.name}*") if path not in log_paths
-                    )
-                except IndexError:
-                    self.target.log.warning("Unexpected NGINX log configuration: %s (%s)", line, config_file)
+            elif (config_file := self.target.fs.path(config_file)).exists() and config_file not in seen:
+                self.parse_config(config_file)
+                seen.add(config_file)
+
+    def parse_config(self, path: Path, seen: set[Path] | None = None) -> None:
+        """Parse the given nginx ``.conf`` file for ``access_log``, ``error_log`` and ``include`` directives."""
 
-        return log_paths
+        seen = set() if seen is None else seen
 
-    @plugin.export(record=WebserverAccessLogRecord)
+        if path in seen:
+            self.target.log.warning("Detected recursion in Nginx configuration, file already parsed: %s", path)
+            return
+
+        seen.add(path)
+
+        if not path.is_file():
+            self.target.log.warning("File %s does not exist on target", path)
+            return
+
+        for line in path.open("rt").readlines():
+            if not (line := line.strip()):
+                continue
+
+            if "access_log " in line:
+                if access_log := RE_ACCESS_LOG_DIRECTIVE.search(line):
+                    access_log = self.target.fs.path(access_log["path"])
+                    self.access_paths.update(p for p in access_log.parent.glob(f"{access_log.name}*"))
+                else:
+                    self.target.log.warning("Unable to parse nginx access_log line %r in %s", line, path)
+
+            elif "error_log " in line:
+                if error_log := RE_ERROR_LOG_DIRECTIVE.search(line):
+                    error_log = self.target.fs.path(error_log["path"])
+                    self.error_paths.update(p for p in error_log.parent.glob(f"{error_log.name}*"))
```

Comment:
```suggestion
                    self.error_paths.update(error_log.parent.glob(f"{error_log.name}*")
```
Wouldn't this work? It's a useless generator comprehension now.